File: Practicas/Practicas completas/Practica_03/a2253320602_Practica03_tarea01.java
package Practica_03;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class a2253320602_Practica03_tarea01 {
    public static void main(String[] args) throws IOException {
        BufferedReader entrada = new BufferedReader(new InputStreamReader(System.in));
        
        System.out.println("Introduce la longitud total del muro (X):");
        double X = Double.parseDouble(entrada.readLine());
        
        System.out.println("Introduce la altura del muro (Y):");
        double Y = Double.parseDouble(entrada.readLine());
        
        System.out.println("Introduce el numero de castillos (N):");
        double N = Double.parseDouble(entrada.readLine());
        
        System.out.println("Introduce la longitud de cada castillo (P):");
        double P = Double.parseDouble(entrada.readLine());
        
        System.out.println("Introduce la longitud del ladrillo:");
        double Lladrillo = Double.parseDouble(entrada.readLine());
        
        System.out.println("Introduce la altura del ladrillo:");
        double Aladrillo = Double.parseDouble(entrada.readLine());
        
        System.out.println("Introduce el espesor de la junta vertical:");
        double Ajuntav = Double.parseDouble(entrada.readLine());
        
        System.out.println("Introduce el espesor de la junta horizontal:");
        double Anjuntah = Double.parseDouble(entrada.readLine());
        
        double longitudEfectiva = X - (N * P);
        double areaMuro = longitudEfectiva * Y;
        double areaLadrillo = (Aladrillo + Anjuntah) * (Lladrillo + Ajuntav);
        double NL = areaMuro / areaLadrillo;
        
        System.out.println("El numero de ladrillos necesarios es: " + NL);
    }
}

File: Practicas/Practicas completas/Practica_03/a2253320602_Practica03_tarea01.py
X = float(input("Introduce la longitud total del muro (X): "))
Y = float(input("Introduce la altura del muro (Y): "))
N = float(input("Introduce el numero de castillos (N): "))
P = float(input("Introduce la longitud de cada castillo (P): "))
Lladrillo = float(input("Introduce la longitud del ladrillo: "))
Aladrillo = float(input("Introduce la altura del ladrillo: "))
Ajuntav = float(input("Introduce el espesor de la junta vertical: "))
Anjuntah = float(input("Introduce el espesor de la junta horizontal: "))

longitud_efectiva = X - (N * P)
area_muro = longitud_efectiva * Y
area_ladrillo = (Aladrillo + Anjuntah) * (Lladrillo + Ajuntav)
NL = area_muro / area_ladrillo

print("El numero de ladrillos necesarios es:", NL)